src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java

```diff
@@ -508,24 +512,30 @@ public static void printRootCauseStackTrace(final Throwable throwable, final Pri
      * @return an array of stack trace frames, never null
      * @since 2.0
      */
+    @SuppressWarnings({"assignment.type.incompatible","array.access.unsafe.low"})/*
+    #1 getThrowables(throwable) returns an array of minimum length 1 if throwable != null
+    #2 i != 0 => i - 1 >= 0
+    #3 i > 0 inside the loop
+    #4 i > 0 inside the loop
+    */ 
     public static String[] getRootCauseStackTrace(final Throwable throwable) {
         if (throwable == null) {
             return ArrayUtils.EMPTY_STRING_ARRAY;
         }
-        final Throwable throwables[] = getThrowables(throwable);
+        final Throwable @MinLen(1) [] throwables = getThrowables(throwable); // #1
         final int count = throwables.length;
         final List<String> frames = new ArrayList<>();
         List<String> nextTrace = getStackFrameList(throwables[count - 1]);
         for (int i = count; --i >= 0;) {
             final List<String> trace = nextTrace;
             if (i != 0) {
-                nextTrace = getStackFrameList(throwables[i - 1]);
+                nextTrace = getStackFrameList(throwables[i - 1]); // #2
```

Comment:
I'm surprised the checker can't figure this one out. Can you turn this into a minimal test case and submit it as a bug to the issue tracker?